create_commits.py

```diff
@@ -0,0 +1,347 @@
+
+
+class FileDeletion(TypedDict):
+    """
+    A type hint for the FileDeletion GraphQL input object, which is an input to the
+    FileChanges GraphQL input object.
+    https://docs.github.com/en/graphql/reference/input-objects#filedeletion
+    """
+
+    path: str
+
+
+class FileAddition(TypedDict):
+    """
+    A type hint for the FileAddition GraphQL input object, which is an input to the
+    FileChanges GraphQL input object.
+    https://docs.github.com/en/graphql/reference/input-objects#fileaddition
+    """
+
+    path: str
+    contents: str
+
+
+class FileChanges(TypedDict):
+    """
+    A type hint for the FileChanges GraphQL input object, which is an input to the
+    createCommitOnBranch mutation.
+    Read about the mutation here:
+    https://docs.github.com/en/graphql/reference/mutations#createcommitonbranch
+    And the file changes input object here:
+    https://docs.github.com/en/graphql/reference/input-objects#filechanges
+    """
+
+    additions: list[FileAddition]
+    deletions: list[FileDeletion]
+
+
+class CommitMessage(TypedDict):
+    """
+    A type hint for the CommitMessage GraphQL input object, which is an input to the
+    createCommitOnBranch mutation.
+    Read about the mutation here:
+    https://docs.github.com/en/graphql/reference/mutations#createcommitonbranch
+    And the commit message input object here:
+    https://docs.github.com/en/graphql/reference/input-objects#commitmessage
+    """
+
+    body: str
+    headline: str
+
+
+def get_file_changes_from_local_commit_hash(commit_hash: str) -> FileChanges:
+    """
+    Create a file changes object for a specific commit.
+
+    Args:
+        commit_hash (str): The hash of the commit.
+
+    Returns:
+        dict: A dictionary representing the FileChanges object.
+    """
+
+    # setting config variable advice.detachedHead to false to avoid verbose messages in the logs
+    subprocess.run(
+        ["git", "config", "advice.detachedHead", "false"],
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+
+    logging.debug("")
+    logging.debug("now working on commit hash %s", commit_hash)
+    # Get a list of files changed in a specific commit.
+    result = subprocess.run(
+        ["git", "diff", "--name-status", f"{commit_hash}~1", commit_hash],
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+    files_changed_by_commit = result.stdout.splitlines()
+
+    logging.debug("files_changed_by_commit:\n %s", files_changed_by_commit)
+    file_changes: FileChanges = {
+        "additions": [],
+        "deletions": [],
+    }
+
+    # check out that commit in detached head state, so that we can pull file contents accurately
+    subprocess.run(["git", "checkout", commit_hash], check=True)
+    logging.debug("output of ls after checking out commit hash:")
+    subprocess.run(["ls"], check=True)
+
+    for line in files_changed_by_commit:
+        status, *filenames = line.split("\t")
```

Comment:
iirc there's a way for the line to have a leading space in this output - what happens in that case?

Reply:
discussed over slack, we couldn't find any examples of this and don't think it's likely